complex-types.d.ts

```diff
@@ -4,7 +4,7 @@ import type {
   VisitorResult,
   Matches,
   InclusiveDescendant
-} from 'unist-util-visit-parents/complex-types'
```

Comment:
Does this really work? I believe some of these types are not re-exported from `visit-parents/index.js` (types with generics can’t be re-exported)

Reply:
It works if done like so:

https://github.com/syntax-tree/unist-util-visit-parents/blob/385428809897e90c83f6f565ee699d6c1667e58c/index.js#L13-L16

The alternative is to add `complex-types.d.ts` to `exports`, what's your preference?

Reply:
Oh, sick... That’s new, I think?

Still, `InclusiveDescendant` and everything else is dropping its generics.

The alternative is also the previous state (+ `.js`)? We don’t use `exports`, so everything is exported already.
A lot of packages depending on this are already using `./complex-types`, so it’s important not to break that. In which case, re-exporting like that is superfluous now?

Reply:
> Still, `InclusiveDescendant` and everything else is dropping its generics.

You're right, I didn't pay attention, thanks! It worked only because they have defaults. And re-exporting it is more complicated than `Matches`... Will fix this.

> The alternative is also the previous state (+ `.js`)? We don’t use `exports`, so everything is exported already.
A lot of packages depending on this are already using `./complex-types`, so it’s important not to break that. In which case, re-exporting like that is superfluous now?

I don't know why, but appending `.js` didn't work for me when I was patching `node_modules` in an existing project of mine. But now I did it again, and it worked fine. So I'll go ahead and undo the re-exports, cheers!